frontend/www/scripts/workouts.js

```diff
@@ -37,6 +37,47 @@ function createWorkout() {
     window.location.replace("workout.html");
 }
 
+function filterWorkouts(searchValue, authorFilter, workouts, currentUser) {
+    let workoutAnchors = document.querySelectorAll('.workout');
+    for (let j = 0; j < workouts.length; j++) {
+        // I'm assuming that the order of workout objects matches
+        // the other of the workout anchor elements. They should, given
+        // that I just created them.
+        let workout = workouts[j];
+        let workoutAnchor = workoutAnchors[j];
+        if (workout.name.toLowerCase().includes(searchValue.toLowerCase())) {
+            switch (authorFilter) {
+                case "list-my-workouts-list":
+                    if (workout.owner == currentUser.url ) {
+                        workoutAnchor.classList.remove('hide');
+                    } else {
+                        workoutAnchor.classList.add('hide');
+                    }
+                    break;
+                case "list-athlete-workouts-list":
+                    if (currentUser.athletes && currentUser.athletes.includes(workout.owner)) {
+                        workoutAnchor.classList.remove('hide');
+                    } else {
+                        workoutAnchor.classList.add('hide');
+                    }
+                    break;
+                case "list-public-workouts-list":
+                    if (workout.visibility == "PU") {
+                        workoutAnchor.classList.remove('hide');
+                    } else {
+                        workoutAnchor.classList.add('hide');
+                    }
+                    break;
+                default :
+                    workoutAnchor.classList.remove('hide');
+                    break;
+            }
+        } else {
+            workoutAnchor.classList.add('hide')
```

Comment:
Small issue with out styling when hiding elements like this: we use Bootstrap for styling the workout list, which adds borders to each element in the list, but only a top border to the first element. When hiding the first element with this, the "new" first element does not get that top border (since the old first element is still in the DOM, just hidden). Honestly I think the best fix would be to just add a top border to every element in the workout list, though I'm not sure how to do that with Bootstrap. This isn't a big issue though.